source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/overview/products/item/tasks.tsx

```diff
@@ -13,70 +13,137 @@ import { useCurrentUser } from '~/data/containers/with-current-user';
 
 import { useUserTaskHelpers } from '~/data/containers/resources/user-task';
 
+import { get as authenticatedGet } from '@lib/fetch';
+
+import { handleResponse } from '~/data/containers/with-current-user/fetcher';
+
+import { RectLoader } from '~/ui/components/loaders';
+
 interface IProps {
   product: ProductResource;
 }
-
 export default function ProductTasksForCurrentUser({ product }: IProps) {
   const { t } = useTranslations();
   const { currentUser } = useCurrentUser();
   const { dataStore } = useOrbit();
+  const productRemoteId = remoteIdentityFrom(dataStore, product).keys.remoteId;
+  const [fetchComplete, setFetchComplete] = useState(false);
+  const [transition, setTransition] = useState();
 
-  let tasks = [];
-
-  try {
-    tasks = dataStore.cache.query((q) =>
-      q
-        .findRecords('userTask')
-        .filter({ relation: 'product', record: product })
-        .filter({ relation: 'user', record: currentUser })
-    );
-
-    // for some reason the backend is sending task creations to the frontend qucik
-    // enough where the key-checking mechanisms hasn't yet added the first task,
-    // so a subsequent task will become a duplicate because both received tasks
-    // have yet to be added to the local cache.
-    tasks = uniqBy(tasks, (task) => (task.keys || {}).remoteId);
-  } catch (e) {
-    console.debug(
-      'error occurred',
-      e,
-      'we probably need to PR to orbit.js, as this is a race condition scenario that causes the error',
-      'variables that are in use that may be related to this error: ',
-      ['product: ', product, 'currentUser: ', currentUser]
-    );
-  }
+  useEffect(() => {
```

Comment:
instead of using useEffect, I think you can define a callback with useCallback and pass that to AsyncWaiter (search around for other usages), it'll also give you loading and error states :)

Reply:
Done